AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -836,7 +845,7 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
 {
   // Passed constrainedSize is the the same as the node's current constrained size it's a noop
   ASDisplayNodeAssertMainThread();
-  if ([self shouldCalculateLayoutWithConstrainedSize:constrainedSize parentSize:constrainedSize.max] == NO) {
+  if (_calculatedDisplayNodeLayout->isValidForConstrainedSizeParentSize(constrainedSize, constrainedSize.max)) {
```

Comment:
@maicki ^ on my previous comment, since this check should never be triggered if -setNeedsLayout is called... should this method also have setNeedsLayout in it and remove the check?  If not, why does this method not have setNeedsLayout called even though one of its main callers does do setNeedsLayout?

Need a comment about the case where this check is important since it's not clear based on the setNeedsLayout...